back/app/controllers/web_api/v1/initiatives_controller.rb

```diff
@@ -136,7 +136,7 @@ def update
     ActiveRecord::Base.transaction do
       saved = @initiative.save save_options
       if saved
-        authorize @initiative
+        authorize @initiative unless @initiative.anonymous? # Cannot auth the user if updated to anonymous
```

Comment:
I don't remember so well why we authorize again after saving, but I think it would be better to move this condition inside the policy instead. I suspect this was to prevent author changes by normal users.

If a user, would set anonymous to true and change the author, the endpoint would accept the update I think while it actually shouldn't. Luckily, the author will be cleared, but we're relying on the order of the conditionals in back/app/models/concerns/anonymous_participation.rb:19-23 for this. If the conditionals would swap in order, suddenly someone will be able to post an initiative as someone else. That's why I think we still want to keep authorization through the policy as an extra safety.

Reply:
Probably worth us pairing on this one. The trouble I had was that the policy gets triggered several times during the save, but by the time it comes to authorizing here there is no trace of which user originally owned the initiative or even if anonymous has changed.

Reply:
I've just commented this out and added a test for updating the author which still fails and all the other tests pass, so I'm not sure whether this line serves any purpose at all. Lets go through it on Monday.